numpy-1.16.3 fails to build on pypy3 win32

Building numpy-1.16.3 from source for pypy3.6-v7.1.1-win32 or pypy3.5-v7.0.0-win32 using Visual Studio 2015/2017 fails with a link error:
```
typeinfo.obj : error LNK2019: unresolved external symbol _PyStructSequence_InitType2 referenced in function _typeinfo_init_structsequences
build\lib.win32-3.6\numpy\core\_multiarray_umath.pypy3-71-win32.pyd : fatal error LNK1120: 1 unresolved externals
```

Probably related to https://github.com/numpy/numpy/pull/13346

This fix works for me:
```diff
diff --git a/numpy/core/src/multiarray/typeinfo.c b/numpy/core/src/multiarray/typeinfo.c
index bc4147841..aa4dd055e 100644
--- a/numpy/core/src/multiarray/typeinfo.c
+++ b/numpy/core/src/multiarray/typeinfo.c
@@ -105,7 +105,7 @@ PyArray_typeinforanged(
 }

 /* Backport, only needed here */
-#if PY_VERSION_HEX < 0x03040000
+#if (PY_VERSION_HEX < 0x03040000) || defined(PYPY_VERSION)
     static int
     PyStructSequence_InitType2(PyTypeObject *type, PyStructSequence_Desc *desc) {
         PyStructSequence_InitType(type, desc);
```

--- a/numpy/core/src/multiarray/typeinfo.c
+++ b/numpy/core/src/multiarray/typeinfo.c
@@ -104,8 +104,10 @@ PyArray_typeinforanged(
     return entry;
 }
 
-/* Backport, only needed here */
-#if PY_VERSION_HEX < 0x03040000
+/* Python version only needed for backport to 2.7 */
+#if (PY_VERSION_HEX < 0x03040000) \
+    || (defined(PYPY_VERSION_NUM) && (PYPY_VERSION_NUM < 0x07020000))
+
     static int
     PyStructSequence_InitType2(PyTypeObject *type, PyStructSequence_Desc *desc) {
         PyStructSequence_InitType(type, desc);

